diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index 990e74712874..7a47c13f532b 100755
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -277,6 +277,13 @@ DTC_FLAGS += -q
 ifeq ($(KBUILD_ENABLE_EXTRA_GCC_CHECKS),)
 DTC_FLAGS += -Wno-unit_address_vs_reg
 endif
+ifeq ($(DTC_EXT),)
+DTC_FLAGS += -Wno-simple_bus_reg \
+	-Wno-unit_address_format \
+	-Wno-pci_bridge \
+	-Wno-pci_device_bus_num \
+	-Wno-pci_device_reg
+endif
 
 ifeq ($(KBUILD_ENABLE_EXTRA_GCC_CHECKS),2)
 DTC_FLAGS += -Wnode_name_chars_strict \
@@ -303,7 +310,11 @@ cmd_dt_S_dtb=						\
 $(obj)/%.dtb.S: $(obj)/%.dtb
 	$(call cmd,dt_S_dtb)
 
-DTC = $(srctree)/tools/dtc
+ifneq ($(DTC_EXT),)
+DTC = $(DTC_EXT)
+else
+DTC = $(objtree)/scripts/dtc/dtc
+endif
 
 quiet_cmd_dtc = DTC     $@
 cmd_dtc = mkdir -p $(dir ${dtc-tmp}) ; \
diff --git a/scripts/dtc/Makefile b/scripts/dtc/Makefile
index a0a523af92a4..f8dc39887762 100755
--- a/scripts/dtc/Makefile
+++ b/scripts/dtc/Makefile
@@ -2,6 +2,9 @@
 # scripts/dtc makefile
 
 hostprogs-y	:= dtc
+ifeq ($(DTC_EXT),)
+always		:= $(hostprogs-y)
+endif
 
 dtc-objs	:= dtc.o flattree.o fstree.o data.o livetree.o treesource.o \
 		   srcpos.o checks.o util.o
